solr/modules/ltr/src/java/org/apache/solr/ltr/model/MultipleAdditiveTreesModel.java

```diff
@@ -303,6 +329,27 @@ private static float scoreNode(float[] featureVector, RegressionTreeNode regress
     }
   }
 
+  private static float scoreNodeWithMissingBranch(float[] featureVector, RegressionTreeNode regressionTreeNode) {
+    while (true) {
+      if (regressionTreeNode.isLeaf()) {
+        return regressionTreeNode.value;
+      }
+      // unsupported feature (tree is looking for a feature that does not exist)
+      if ((regressionTreeNode.featureIndex < 0)
+              || (regressionTreeNode.featureIndex >= featureVector.length)) {
+        return 0f;
+      }
+
+      if ((featureVector[regressionTreeNode.featureIndex] <= regressionTreeNode.threshold) ||
```

Comment:
I am wondering if we really needs here the duplicate method again, given we are now using all float[].
In case we are in the scenario of "nullIsSameAsZero" we will never have "Nan", only zeroes, so the first condition is always true and we don't evaluate the second, then we can check the >0 and only in the end do the nan check.
e.g.
`public RegressionTreeNode getMissing() {
      switch (missing) {
        case "left":
          return left;
        default:
          return right;
      }
    }` at the RegressionTreeNode level.
    
    And then:
    `if ((featureVector[regressionTreeNode.featureIndex] <= regressionTreeNode.threshold)) {
        regressionTreeNode = regressionTreeNode.left;
      } else if ((featureVector[regressionTreeNode.featureIndex] > regressionTreeNode.threshold)) {
        regressionTreeNode = regressionTreeNode.right;
      } else if (Float.isNaN(featureVector[regressionTreeNode.featureIndex])) {
        regressionTreeNode = regressionTreeNode.getMissing();
      }`
      
      Potentially we can only make conditional this bit of the method?